File: examples/thrift-server/scala/src/main/scala/com/twitter/finatra/example/CalculatorServer.scala
package com.twitter.finatra.example

import com.twitter.finatra.thrift.ThriftServer
import com.twitter.finatra.thrift.routing.ThriftRouter
import com.twitter.finatra.thrift.filters._

object CalculatorServerMain extends CalculatorServer

class CalculatorServer extends ThriftServer {
  override val name = "calculator-server"

  override def configureThrift(router: ThriftRouter): Unit = {
    router
      .filter[LoggingMDCFilter]
      .filter[TraceIdMDCFilter]
      .filter[ThriftMDCFilter]
      .filter[AccessLoggingFilter]
      .filter[StatsFilter]
      .filter[ExceptionMappingFilter]
      .add[CalculatorController]
  }
}

File: examples/thrift-server/java/src/main/java/com/twitter/finatra/example/CalculatorServer.java
package com.twitter.finatra.example;

import java.util.Collection;
import java.util.Collections;

import com.google.inject.Module;

import com.twitter.finatra.example.modules.ModeModule;
import com.twitter.finatra.thrift.AbstractThriftServer;
import com.twitter.finatra.thrift.filters.AccessLoggingFilter;
import com.twitter.finatra.thrift.filters.LoggingMDCFilter;
import com.twitter.finatra.thrift.filters.StatsFilter;
import com.twitter.finatra.thrift.filters.ThriftMDCFilter;
import com.twitter.finatra.thrift.filters.TraceIdMDCFilter;
import com.twitter.finatra.thrift.routing.JavaThriftRouter;

class CalculatorServer extends AbstractThriftServer {

  @Override
  public Collection<Module> javaModules() {
    return Collections.singletonList(new ModeModule());
  }

  @Override
  public void configureThrift(JavaThriftRouter router) {
    router
        .filter(LoggingMDCFilter.class)
        .filter(TraceIdMDCFilter.class)
        .filter(ThriftMDCFilter.class)
        .filter(AccessLoggingFilter.class)
        .filter(StatsFilter.class)
        .add(CalculatorController.class);
  }
}
